File: pyjl/tests/tests_new/list_op_test.py

def multiply_list(elems):
    new_elems = []
    for e in elems:
        new_elems.append(e*2)
    return new_elems

if __name__ == "__main__":
    a = list()

    # Append
    a.append("test")
    assert a == ["test"]

    # Clear
    a.clear()
    assert a == []
    assert len(a) == 0

    # Remove
    a.append("test1")
    a.append("test2")
    a.remove("test1")
    assert a == ["test2"]
    assert len(a) == 1
    a.clear() # reset list

    # Copy
    a.append("test")
    b = a.copy()
    assert b == a
    a.clear() # reset list

    # Count
    a.append("test2")
    a.append("test2")
    a.remove("test2")
    assert a.count("test2") == 1
    a.clear() # reset list

    # Extend
    a.append("test1")
    a.extend(b)
    assert a == ["test1", "test"]
    a.clear() # reset list

    # Limitation
    elems = ["1", "2", "3"]
    new_elems = multiply_list(elems)
    assert new_elems == ["11", "22", "33"]

    print("OK")

    # Index
    # a.append("test")
    # assert a.index("test") == 0 # Problems

File: pyjl/tests/tests_new-py2many/list_op_test.jl
function multiply_list(elems)::Vector
new_elems = []
for e in elems
push!(new_elems, e*2);
end
return new_elems
end

function main()
a = Vector()
push!(a, "test");
@assert(a == ["test"])
empty!(a);
@assert(a == [])
@assert(length(a) == 0)
push!(a, "test1");
push!(a, "test2");
a = deleteat!(a, findfirst(isequal("test1"), a));
@assert(a == ["test2"])
@assert(length(a) == 1)
empty!(a);
push!(a, "test");
b = copy(a)
@assert(b == a)
empty!(a);
push!(a, "test2");
push!(a, "test2");
a = deleteat!(a, findfirst(isequal("test2"), a));
@assert(count(isequal("test2"), a) == 1)
empty!(a);
push!(a, "test1");
a = append!(a, b);
@assert(a == ["test1", "test"])
empty!(a);
elems = ["1", "2", "3"]
new_elems = multiply_list(elems)
@assert(new_elems == ["11", "22", "33"])
println("OK");
end

main()
